host/cli/discover.go

```diff
@@ -0,0 +1,185 @@
+package cli
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+	"text/tabwriter"
+	"text/template"
+
+	"github.com/flynn/flynn/Godeps/_workspace/src/github.com/flynn/go-docopt"
+	"github.com/flynn/flynn/discoverd/client"
+	"github.com/flynn/flynn/pkg/cluster"
+)
+
+func init() {
+	Register("discover", runDiscover, `
+usage: flynn-host discover [options] <service>...
+
+Return low-level information about a service.
+
+Options:
+  --json                 Pretty print response as JSON
+  -f, --format=<format>  Format the output using the given go template
+
+Examples:  
+
+  Show information for service 'flynn-host'.
+
+  $ flynn-host discover flynn-host
+  SERVICE     ID                                ADDR
+  flynn-host  8d0d15d0b14d9efe6316493a658e6b3f  10.0.2.15:1113
+  
+  $ flynn-host discover --json controller
+  [
+    {
+      "Name": "controller",
+      "Leader": {
+        "id": "0bbd29d630f63b11ae999b1dabd565f9",
+        "addr": "100.100.36.4:80",
+        "proto": "http",
+        "meta": {
+          "AUTH_KEY": "9b7b3b382f13425da3a8cb390f0937b8",
+          "FLYNN_APP_ID": "2169984c-c15f-499d-be77-636612425330",
+          "FLYNN_JOB_ID": "host-b773161b-3e50-43ec-bb38-5c492c2ba2fa",
+          "FLYNN_PROCESS_TYPE": "web",
+          "FLYNN_RELEASE_ID": "305a5c6b-7f01-4232-bc1c-74b6a408960d"
+        },
+        "index": 22
+      },
+      "Instances": [
+        {
+          "id": "0bbd29d630f63b11ae999b1dabd565f9",
+          "addr": "100.100.36.4:80",
+          "proto": "http",
+          "meta": {
+            "AUTH_KEY": "9b7b3b382f13425da3a8cb390f0937b8",
+            "FLYNN_APP_ID": "2169984c-c15f-499d-be77-636612425330",
+            "FLYNN_JOB_ID": "host-b773161b-3e50-43ec-bb38-5c492c2ba2fa",
+            "FLYNN_PROCESS_TYPE": "web",
+            "FLYNN_RELEASE_ID": "305a5c6b-7f01-4232-bc1c-74b6a408960d"
+          },
+          "index": 22
+        }
+      ],
+      "Meta": {
+        "Data": {},
+        "index": 0
+      }
+    }
+  ]
+  
+  Show id and address of every 'flynn-host' service instance.
+  
+  $ flynn-host discover --json -f '{{range .Instances}}{{index .Meta "id"}} {{.Addr}}{{end}}' flynn-host
+  host 10.0.2.15:1113
```

Comment:
The behavior should be this:

- Error if `--json` and `--format` are provided at the same time.
- If no flags provided, print the table.
- Otherwise do the JSON or formatted output based on the flag.